Domain/PocketContainerExtensions.cs

```diff
@@ -10,20 +10,39 @@ internal static class PocketContainerExtensions
     {
         public static PocketContainer UseImmediateCommandScheduling(this PocketContainer container)
         {
-            return container.AddStrategy(type =>
+            container.AddStrategy(type =>
             {
                 if (type.IsInterface &&  
                     type.IsGenericType && 
                     type.GetGenericTypeDefinition() == typeof(ICommandScheduler<>))
                 {
-                    var aggregateType = type.GetGenericArguments().First();
-                    var schedulerType = typeof (CommandScheduler<>).MakeGenericType(aggregateType);
+                    var targetType = type.GetGenericArguments().First();
+                    var schedulerType = typeof (CommandScheduler<>).MakeGenericType(targetType);
 
                     return c => c.Resolve(schedulerType);
                 }
 
                 return null;
             });
+
+            container.AddStrategy(type =>
+            {
```

Comment:
I think this deserves a separate method, since it's orthogonal to immediate scheduling. That method should probably be invoked in the constructor for `Configuration` since I suspect we want it to be a default behavior.